File: notification-service/src/main/java/com/eirsteir/coffeewithme/notification/dto/NotificationDto.java
package com.eirsteir.coffeewithme.notification.dto;

import com.eirsteir.coffeewithme.commons.domain.notification.NotificationType;
import com.eirsteir.coffeewithme.notification.domain.Notification;
import java.util.Date;
import lombok.*;
import lombok.experimental.Accessors;

@Getter
@Setter
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@Accessors(chain = true)
public class NotificationDto {

  private Long notificationId;
  private Date timestamp;
  private Notification.UserDetails user;
  private Boolean seen;
  private NotificationType type;
}

File: notification-service/src/main/kotlin/com/eirsteir/coffeewithme/notification/dto/NotificationDto.kt
package com.eirsteir.coffeewithme.notification.dto

import com.eirsteir.coffeewithme.commons.domain.notification.NotificationType
import com.eirsteir.coffeewithme.notification.domain.Notification
import lombok.*
import lombok.experimental.Accessors
import java.util.*


data class NotificationDto(
    val notificationId: Long,
    val timestamp: Date? = null,
    val user: Notification.UserDetails? = null,
    val type: NotificationType? = null,
    val seen: Boolean? = null
)